clang/lib/3C/AVarBoundsInfo.cpp

```diff
@@ -68,8 +68,7 @@ bool AVarBoundsInfo::isValidBoundVariable(clang::Decl *D) {
   }
   if (VarDecl *VD = dyn_cast<VarDecl>(D)) {
     return !VD->getNameAsString().empty();
-  }
-  if (FieldDecl *FD = dyn_cast<FieldDecl>(D)) {
+  } else if (FieldDecl *FD = dyn_cast<FieldDecl>(D)) {
```

Comment:
Why add the `else` ? I guess it's impossible for `D` to be both, so it's fine?

Reply:
There were cases where it could be both. We should give preference to `VarDecl`.